recipes/adios2/meta.yaml

```diff
@@ -0,0 +1,106 @@
+{% set name = "adios2" %}
+{% set version = "2.4.0" %}
+{% set build = 0 %}
+{% set sha256 = "50ecea04b1e41c88835b4b3fd4e7bf0a0a2a3129855c9cc4ba6cf6a1575106e2" %}
+
+# ensure mpi is defined (needed for conda-smithy recipe-lint)
+{% set mpi = mpi or 'nompi' %}
+
+# prioritize nompi variant via build number
+{% if mpi == 'nompi' %}
+{% set build = build + 100 %}
+{% endif %}
```

Comment:
This doesn't seem like the right way to do this.

Reply:
Not? That's how @minrk showed it to me. We increase the build number for nompi variants to select them by default.

Reply:
Really?  That is kinda nuts.

Reply:
We don't have documentation for this do we?

Reply:
Somewhat, I am just following recently explored best practices:
- https://hackmd.io/ry4uI0thTs2q_b4mAQd_qg
- https://github.com/conda-forge/adios-feedstock/pull/12#discussion_r245610080
- https://github.com/conda-forge/hdf5-feedstock/blob/master/recipe/meta.yaml
- https://github.com/conda-forge/h5py-feedstock/blob/master/recipe/meta.yaml

Sounds actually quite legit to me

Reply:
I think this way of doing things is causing issues in the solver with multi output recipes that depend on packages with MPI variants. I've not managed to make sense of it yet but these builds show it:
* https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=45775
* https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=45278

Reply:
> We don't have documentation for this do we?

It's [in the docs](https://conda-forge.org/docs/maintainer/knowledge_base.html#message-passing-interface-mpi)

Build offset is the only way to indicate preference in the solver that I'm aware of that works for existing packages. Note that the original and first-implemented plan was to use track_features to deprioritize, but this does not work for packages that have ever been published before, but I think it might work for *new* packages, or packages where all previous builds have been fully revoked or hotfixed.

In most cases, packages that require MPI express no preference, but it's more complicated when you have three variants and want users to opt-in to MPI (e.g. hdf5). Conda still has no ability to express preferences in a natural way, so any solution is going to be a bit wonky, or